packages/context/src/Callback/Callback.container.tsx

```diff
@@ -6,15 +6,15 @@ import withServices from '../withServices';
 export const onRedirectSuccess = (history: ReactOidcHistory, oidcLogInternal: typeof oidcLog) => (user: User) => {
   oidcLogInternal.info('Successfull login Callback', user);
   if (user.state.url) {
-    history.push(user.state.url);
+    history.replaceCurrent(user.state.url);
   } else {
     oidcLogInternal.warn('no location in state');
   }
 };
 
-export const onRedirectError = (history: ReactOidcHistory, oidcLogInternal: typeof oidcLog) => ({ message }: { message: string }) => {
+export const onRedirectError = (oidcLogInternal: typeof oidcLog, userManager: UserManager) => ({ message }: { message: string }) => {
   oidcLogInternal.error(`There was an error handling the token callback: ${message}`);
-  history.push(`/authentication/not-authenticated?message=${encodeURIComponent(message)}`);
+  userManager.signinRedirect({ data: { url: "/" } })
```

Comment:
I have made some tests it works pretty well.
But you cannot set a default behavior on error. It won't work with all use cases.

Reply:
Exactly. This is why I had this line to handle missing state only:
```javascript
if (message === "No matching state found in storage") {
```
I don't think there is another way to detect that state is missing in the state store.
Do you want to me revert to the previous version of the fix?

Reply:
@guillaume-chervet, do you have any feedback on this?

Reply:
Sorry again for the delay.

I cannot accept the pull request with that line. I think you might remove your change in onRedirectError.
I think you cannot set up this default behavior for every use case. 
You can set it up on your own application by using this event.

What do you think ?

Reply:
What do you think about this : #627